RAT/classlist.py

```diff
@@ -0,0 +1,202 @@
+    def __setitem__(self, index: int, set_dict: Dict[str, Any]) -> None:
+        """Assign the values of an existing object's attributes using a dictionary."""
+        self._validate_name(set_dict)
+        for key, value in set_dict.items():
+            setattr(self.data[index], key, value)
+
+    def __iadd__(self, other: Iterable[object]) -> 'ClassList':
+        """Define in-place addition using the "+=" operator."""
+        self._check_classes(self + other)
+        self._check_unique_names(self + other)
+        super().__iadd__(other)
+        return self
+
+    def __mul__(self, n: int) -> None:
+        """Define multiplication using the "*" operator."""
+        raise TypeError(f"unsupported operand type(s) for *: '{self.__class__.__name__}' and '{n.__class__.__name__}'")
+
+    def __rmul__(self, n: int) -> None:
+        """Define multiplication using the "*" operator."""
+        raise TypeError(f"unsupported operand type(s) for *: '{n.__class__.__name__}' and '{self.__class__.__name__}'")
+
+    def __imul__(self, n: int) -> None:
+        """Define in-place multiplication using the "*=" operator."""
+        raise TypeError(f"unsupported operand type(s) for *=: '{self.__class__.__name__}' and '{n.__class__.__name__}'")
+
+    def append(self, **kwargs) -> None:
+        """Append a new object to the ClassList using keyword arguments to set attribute values."""
+        self._validate_name(kwargs)
+        self.data.append(self._class_handle(**kwargs))
+
+    def insert(self, index: int, **kwargs) -> None:
+        """Insert a new object into the ClassList at a given index using keyword arguments to set attribute values."""
+        self._validate_name(kwargs)
+        self.data.insert(index, self._class_handle(**kwargs))
+
+    def remove(self, item: Union[object, str]) -> None:
+        """Remove an object from the ClassList using either the object itself or its name."""
+        if not isinstance(item, self._class_handle):
+            item = self._get_item_from_name(item)
+        self.data.remove(item)
+
+    def count(self, item: Union[object, str]) -> int:
+        """Return the number of times an object appears in the ClassList using either the object itself or its name."""
+        if not isinstance(item, self._class_handle):
+            item = self._get_item_from_name(item)
+        return self.data.count(item)
+
+    def index(self, item: Union[object, str], *args) -> int:
+        """Return the index of a particular object in the ClassList using either the object itself or its name."""
+        if not isinstance(item, self._class_handle):
+            item = self._get_item_from_name(item)
+        return self.data.index(item, *args)
+
+    def extend(self, other: Iterable[object]) -> None:
+        """Extend the ClassList by adding another iterable."""
+        self._check_classes(self + other)
+        self._check_unique_names(self + other)
+        self.data.extend(other)
+
+    def get_names(self) -> list[str]:
```

Comment:
I am conflicted about this, I don't think there is proper separation of concerns, it is tightly coupled to the data having a name property instead of a variable with a given property. For example, if the unique column was called 'ID' not 'name', this class will have to be modified which is not great

Reply:
Ok, I think I can rewrite this to allow us to set the field we use. Let's see.

Reply:
Yes, that's done and works.